tests/staking/StakingTest.js

```diff
@@ -236,6 +251,435 @@ contract("Staking", (accounts) => {
+                lockedDate,
+                blockBefore
+            );
+            let priorDelegateStakeAfter = await staking.getPriorStakeByDateForDelegatee(
+                delegatee1,
+                lockedDate,
+                blockAfter
+            );
+            expect(priorDelegateStakeBefore.sub(priorDelegateStakeAfter)).to.be.equal(amount1);
+
+            //check getPriorStakeByDateForDelegatee - delegatee2
+            priorDelegateStakeBefore = await staking.getPriorStakeByDateForDelegatee(
+                delegatee2,
+                lockedDate,
+                blockBefore
+            );
+            priorDelegateStakeAfter = await staking.getPriorStakeByDateForDelegatee(
+                delegatee2,
+                lockedDate,
+                blockAfter
+            );
+            expect(priorDelegateStakeAfter.sub(priorDelegateStakeBefore)).to.be.equal(
+                amount1.add(amount2)
+            );
+        });
+
+        it("should update delegate stake after withdraw", async () => {
+            let user = accounts[0];
+            let lockedDate = kickoffTS.add(new BN(TWO_WEEKS).mul(new BN(2)));
+            let amount1 = new BN(1000);
+            let amount2 = new BN(2000);
+            let totalAmount = amount1.add(amount2);
+            await token.transfer(user, totalAmount);
+            await token.approve(staking.address, totalAmount, { from: user });
+
+            await staking.stake(amount1, lockedDate, user, user, { from: user });
+            await staking.withdraw(amount1, lockedDate, user, { from: user });
+
+            let tx = await staking.stake(amount2, lockedDate, user, user, { from: user });
+            let txBlockNumber = new BN(tx.receipt.blockNumber.toString());
+            await mineBlock();
+            let blockBefore = txBlockNumber.sub(new BN(1));
+            let blockAfter = txBlockNumber;
+
+            //check getPriorStakeByDateForDelegatee - delegatee1
+            let priorDelegateStakeBefore = await staking.getPriorStakeByDateForDelegatee(
+                user,
+                lockedDate,
+                blockBefore
+            );
+            let priorDelegateStakeAfter = await staking.getPriorStakeByDateForDelegatee(
+                user,
+                lockedDate,
+                blockAfter
+            );
+            expect(priorDelegateStakeAfter.sub(priorDelegateStakeBefore)).to.be.equal(amount2);
+        });
+
+        // if stakeFor is a vesting contract, getPriorVestingStakeByDate returns the increased vesting stake for stakeFor and until
+        it("should update stake for vesting contracts", async () => {
+            let user = accounts[0];
+            let lockedDate = kickoffTS.add(new BN(TWO_WEEKS).mul(new BN(2)));
+            let amount = new BN(1000);
+            await token.transfer(user, amount);
+            await token.approve(staking.address, amount, { from: user });
+
+            await staking.addContractCodeHash(user);
```

Comment:
in a normal flow vesting contract stakes for a user's address which differs from the vesting contract  
the test covers a sort of a corner case when a vesting contract stakes for itself which probably should revert

Reply:
We already have tests when msg.sender and stakeFor are different.

Reply:
We already have tests when msg.sender and stakeFor are different.